### PR TITLE
Bottom sheet

### PR DESCRIPTION
**Pull request by 3lvis**
artboard

Moved from FilterKit.

### CODE REVIEW

**Comment by 3lvis** [comment]
It doesn't work as well as I would like but it's a start :)

File: Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public final class BottomSheetInteractiveDismissalController: UIPercentDrivenInteractiveTransition {
+    let dismissalTransitioningRect: CGRect
```
Comment (vadymmarkov) [inline Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift]: Could those 4 properties be private? or they're used somewhere outside of this class?

File: Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        case .ended:
+            isInteractionActive = false
+            if dismissalPercentage < 0 {
+                cancel()
+            } else if percentComplete >= dismissalPercentageThreshold {
+                finish()
+            } else {
+                if #available(iOS 11.0, *) {
+                } else {
+                    // There is a bug for version earlier than iOS 11 that needs completion speed to be something else than 1 to avoid "double" cancel dismissal animation
```
Comment (vadymmarkov) [inline Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift]: 😱

File: Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift (hunk 1 above)
Comment (3lvis) [reply Sources/Components/BottomSheet/BottomSheetInteractiveDismissalController.swift]: Good point! I should take some time to go through the access control.

**Comment by 3lvis** [comment]
Closing seeing as this implementation isn't good enough.

**Pull request closed without merging**